### PR TITLE
Improvement: Implemented Feedback from Code Review 

### PR DESCRIPTION
**Pull request by Tarekkk1**

### CODE REVIEW

File: Observer/PurchaseObserver.php [hunk 1]
```diff
@@ -5,46 +5,50 @@
...
         $order = $observer->getEvent()->getOrder();
         $customerEmail = $order->getCustomerEmail();
         $total = $order->getGrandTotal();
+        $currency = $order->getOrderCurrencyCode();
+        $items = $order->getAllVisibleItems();
+        $contents = [];
+        $contentIds = [];
+        foreach ($items as $item) {
+            $contents[] = [
+                'id' => (string)$item->getProductId(),
```
Comment (MostafaShehab) [inline Observer/PurchaseObserver.php:38]: Are you sure that Facebook will accept multiple products in this format? Did you validate this with their docs?

File: Plugin/AddToCartGraphQlPlugin.php [hunk 2]
```diff
@@ -2,56 +2,90 @@
...
+                $cartItem = $cart->getItemByProductSku($sku);
+                if (!$cartItem) {
+                    $this->logger->warning('Cart item not found for SKU: ' . $sku);
+                    continue;
                 }
-            } else {
-                $this->logger->warning('Unexpected cart items format.');
+
+                $customerEmail = '';
+                if ($cart->getCustomer() && $cart->getCustomer()->getEmail()) {
```
Comment (MostafaShehab) [inline Plugin/AddToCartGraphQlPlugin.php:55]: No need to fetch the customer, simply fetch the `customerEmail` directly from the cart. Note that is might be null if this is a guest cart and the user never reached the checkout cycle.

File: Plugin/AddToCartGraphQlPlugin.php [hunk 3]
```diff
@@ -2,56 +2,90 @@
...
+                    $customerEmail = $cart->getCustomer()->getEmail();
+                }
+                
+                $currencyCode = $this->storeManager->getStore()->getCurrentCurrencyCode();
+
+                $eventData = [
+                    'event_name' => 'AddToCart',
+                    'event_time' => time(),
+                    'event_source_url' => (isset($_SERVER['HTTPS']) ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]",
```
Comment (MostafaShehab) [inline Plugin/AddToCartGraphQlPlugin.php:64]: Can you explain what this means?

**Review by MostafaShehab — CHANGES_REQUESTED** [review]

File: Plugin/AddToCartGraphQlPlugin.php [hunk 4]
```diff
@@ -2,56 +2,90 @@
...
+                    'user' => [
+                        'email' => hash('sha256', $customerEmail),
+                    ],
+                    'custom_data' => [
+                        'content_name' => $cartItem->getName(),
+                        'content_id' => $cartItem->getProductId(),
```
[the remark this answers is not in the document]
Comment (Tarekkk1) [reply Plugin/AddToCartGraphQlPlugin.php:70]: Done

File: Plugin/AddToCartGraphQlPlugin.php (hunk 2 above)
Comment (Tarekkk1) [reply Plugin/AddToCartGraphQlPlugin.php:55]: Done

File: Plugin/AddToCartGraphQlPlugin.php (hunk 3 above)
Comment (Tarekkk1) [reply Plugin/AddToCartGraphQlPlugin.php:64]: This is a good way i have seen it while searching that help people to send the full url to any page any place i have used it just to try it as its my first time to see it

Comment (Tarekkk1) [reply Plugin/AddToCartGraphQlPlugin.php:64]: I will remove it or comment it as we dont need it so much

Comment (Tarekkk1) [reply Plugin/AddToCartGraphQlPlugin.php:64]: Screenshot 2023-09-06 065509

Comment (MostafaShehab) [reply Plugin/AddToCartGraphQlPlugin.php]: There is a couple of challenges here. First thing, this is not the standard way in Magento (even though it is the standard in PHP itself). Look into a class called `Request` which comes with Magento, it should have the needed functionality. Second issue, what about mobile applications? They have no url so how will you handle that?

Comment (Tarekkk1) [reply Plugin/AddToCartGraphQlPlugin.php]: Okaaaaayyy got it, I will read about this feature and I have commented it in last modification

File: Observer/PurchaseObserver.php [hunk 5]
```diff
@@ -5,46 +5,52 @@
...
         $order = $observer->getEvent()->getOrder();
         $customerEmail = $order->getCustomerEmail();
         $total = $order->getGrandTotal();
+        $currency = $order->getOrderCurrencyCode();
+        $items = $order->getAllVisibleItems();
+        $contents = [];
+        $contentIds = [];
+        foreach ($items as $item) {
+            $contents[] = [
+                'id' => (string)$item->getSku(),
```
Comment (MostafaShehab) [inline Observer/PurchaseObserver.php:38]: Skus are saved as strings, no need to type cast.

File: Plugin/AddToCartGraphQlPlugin.php [hunk 6]
```diff
@@ -2,56 +2,89 @@
...
+                
+                $cartItem = $cart->getItemByProductSku($sku);
+                if (!$cartItem) {
+                    $this->logger->warning('Cart item not found for SKU: ' . $sku);
+                    continue;
                 }
-            } else {
-                $this->logger->warning('Unexpected cart items format.');
+
+                $customerEmail = '';
```
Comment (MostafaShehab) [inline Plugin/AddToCartGraphQlPlugin.php:54]: This line has no meaning so remove it.

File: Plugin/AddToCartGraphQlPlugin.php [hunk 7]
```diff
@@ -2,56 +2,89 @@
...
+                $customerEmail = $cart->getCustomer()->getEmail();
+               
+                
+                $currencyCode = $this->storeManager->getStore()->getCurrentCurrencyCode();
+
+                $eventData = [
+                    'event_name' => 'AddToCart',
+                    'event_time' => time(),
+                    // 'event_source_url' => (isset($_SERVER['HTTPS']) ? "https" : "http") . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]",
```
Comment (MostafaShehab) [inline Plugin/AddToCartGraphQlPlugin.php:63]: Unless you are leaving this commented line for a reason, it is better to fully remove commented code. Git will track the history for you :D

**Review by MostafaShehab — CHANGES_REQUESTED** [review]

**Review by MostafaShehab — APPROVED** [review]

**Pull request merged**